### PR TITLE
Add error message to projects page

### PR DESCRIPTION
**Pull request by oryxwhite**
This PR adds an error message to the project page that triggers when the fetch request fails.  It includes a try again button that triggers a refetch.  It is covered by a Cypress test.

### CODE REVIEW

**Review by mentor-tara — APPROVED** [review]
Go go go :smile:[](prolog-8)

File: cypress/e2e/project-list.cy.ts [hunk 1]
```diff
@@ -59,3 +59,34 @@ describe("Project List", () => {
...
+      cy.intercept("GET", "https://prolog-api.profy.dev/project").as(
+        "fetchRequest",
+      );
+      cy.wait("@fetchRequest").then((interception) => {
+        expect(interception.request.url).to.equal(
+          "https://prolog-api.profy.dev/project",
+        );
+      });
+    },
+  );
```
Comment (jkettmann) [inline cypress/e2e/project-list.cy.ts:91]: Great job with this test. Just fyi below is the test I wrote which doesn't require intercepting the succeeding request (note the `times: 4`)

```
it("shows an error message", () => {
    // setup request mock
    cy.intercept(
      { url: "https://prolog-api.profy.dev/project", times: 4 },
      {
        statusCode: 500,
      },
    );

    // open issues page
    cy.visit("http://localhost:3000/dashboard/issues");

    // check that the error message is shown
    cy.get("[data-testid='issues-error-message']", { timeout: 15000 })
      .should("be.visible")
      // click retry button
      .find("button")
      .click();

    cy.get("[data-testid='issue-list']").should("be.visible");
  });
```

File: cypress/e2e/project-list.cy.ts [hunk 2]
```diff
@@ -59,3 +59,34 @@ describe("Project List", () => {
...
+
+      cy.wait("@failedFetch");
+
+      cy.get('[data-cy="alert-error"]', { timeout: 12000 }).should(
+        "be.visible",
+      );
+      cy.get('[data-cy="alert-error"]').find("button").click();
+      cy.intercept("GET", "https://prolog-api.profy.dev/project").as(
+        "fetchRequest",
+      );
```
Comment (jkettmann) [inline cypress/e2e/project-list.cy.ts:84]: Clicking the button first and then setting up the intercept might cause this test to fail randomly. At least that's my guess 😄

File: features/projects/components/project-list/project-list.module.scss [hunk 3]
```diff
@@ -15,3 +17,33 @@
...
   }
 }
+
+.error {
+  display: flex;
+  justify-content: space-between;
+  border-radius: 8px;
+  border: 1px solid color.$error-300;
+  background: color.$error-25;
+  padding: 16px;
```
Comment (jkettmann) [inline features/projects/components/project-list/project-list.module.scss:27]: You could use the space variables here. I think it's `space.$s4`!?

**Pull request merged**